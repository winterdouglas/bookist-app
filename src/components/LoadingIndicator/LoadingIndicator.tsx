import { StyleProp, Text, View, ViewStyle } from "react-native";
import { useTranslation } from "react-i18next";

// TODO: Use the proper Text component
// import { Text } from "@/components/Text";

export const LoadingIndicator = () => {
  const { t } = useTranslation();

  return (
    <View style={$containerStyle}>
      <Text>{t("loading")}</Text>
    </View>
  );
};

const $containerStyle: StyleProp<ViewStyle> = {
  flex: 1,
  justifyContent: "center",
  alignItems: "center",
};
