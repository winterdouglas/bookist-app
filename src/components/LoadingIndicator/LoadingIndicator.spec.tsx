import renderer from "react-test-renderer";
import { LoadingIndicator } from "./LoadingIndicator";

it("renders correctly", () => {
  const tree = renderer.create(<LoadingIndicator />).toJSON();
  expect(tree).toMatchSnapshot();
});
