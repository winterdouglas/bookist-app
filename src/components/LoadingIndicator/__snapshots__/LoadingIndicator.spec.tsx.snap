// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "alignSelf": "center",
        "flexDirection": "row",
        "gap": 8,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    size="small"
  />
  <Text
    selectable={true}
    style={
      [
        [
          {
            "fontSize": 16,
          },
          {
            "fontWeight": "500",
          },
        ],
        undefined,
        {
          "color": "rgb(28, 28, 30)",
        },
        undefined,
      ]
    }
  >
    loading
  </Text>
</View>
`;
